python/RobotControl/RunningWithSSH.py

```diff
@@ -16,27 +25,112 @@ def run_script_on_robot(script: str) -> str:
         Returns: 
             An error message or an empty string.
     """
-    robot.interpreter_mode.start()
+    robot.ssh.write_script(script)
+    robot.controller.load_program()
+    robot.controller.start_program()
+    sleep(0.1)
 
-    interpreter_script = "halt\n" + script
-    result = robot.interpreter_mode.send_small_command_on_interpreter(interpreter_script)
-    non_recurring_logger.debug(f"Result of sending script to interpreter: {result}")
+    latest_errors = robot.ssh.read_lines_from_log(2)
 
-    error_message = result.lower().split(":")[1]
-    out = None
+    if "Compile error" in latest_errors or "Lexer exception" in latest_errors:
+        non_recurring_logger.debug("Compile error or Lexer exception found")
+        error = latest_errors.split("\n")[1]
+        parts = re.split(r'ERROR\s+-', error, maxsplit=1)
+        return parts[1]
 
-    if "error" in error_message:
-        out = f"{result.split(":")[1].strip()}: {result.split(":")[2].strip()}"
-    elif "exception" in error_message:
-        out = result.split(":")[1].strip()
+    #Start thread to check for runtime errors
+    def run_async_checker():
+        asyncio.run(run_script_finished_error_checker(id))
 
-    if out:
-        robot.controller.stop_program()
-        return out
+    error_checker_thread = threading.Thread(target=run_async_checker)
+    error_checker_thread.start()
 
-    robot.controller.stop_program()
+    return "" 
 
-    robot.ssh.write_script(script)
-    robot.controller.load_program()
-    robot.controller.start_program()
-    return "" 
\ No newline at end of file
+async def run_script_finished_error_checker(id: int):
+    """
+    Check for errors in the script after it has finished running.
+    Sends the error to all web clients if an error is found.
+
+        Args:
+            script (str): The script to check.
+    
+        Returns: 
+            None
+    """
+    recurring_logger.debug(f"Robot mode: {robot.controller.robot_mode}")
+    __wait_for_condition(lambda: "Starting" not in robot.controller.robot_mode)
+    
+    recurring_logger.debug(f"Program state: {robot.controller.program_state}")
+    __wait_for_condition(lambda: "PLAYING" not in robot.controller.program_state)
+    
+    latest_errors = robot.ssh.read_lines_from_log(3)
+    recurring_logger.debug(f"Latest errors:\n{latest_errors}")
+    if "Type error" in latest_errors or "Runtime error" in latest_errors:
+        error_line = latest_errors.split("\n")[0]
+        error_text = re.split(r'ERROR\s+-', error_line, maxsplit=1)[1]
+
+        error = retrieve_line_number_text(error_text)
+        __send_error_message_to_web_clients(id, error)
+        return
+    
+    if "New safety mode: SAFETY_MODE_PROTECTIVE_STOP" in latest_errors:
+        error_text = """
+        PROTECTIVE STOP: Reconsider how the robot is moving.
+        You will not be able to run a program for five seconds.
+        """
+        __send_error_message_to_web_clients(id, error_text)
```

Comment:
I would think that this throws a ValueError in the ACKResponse constructor because the message text does not include the text: `error`.

The fix i believe is to modify the ACKResponse constructor, to explicitly include a status argument. Even if it does currently work it seems very fragile to me.